### ISSUE TITLE
Cross-instance account linking/syncing? (nomadic identity)

### ISSUE THREAD

**Issue by AlexBlackRaven** [LemmyNet/lemmy-ui#1571]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

Managing accounts from other instances (from your own instance) seems like it would be a useful feature for lemmy. Smaller instances are not as stable as big ones (hosting costs, drama and falling off in popularity can make em fall off sometimes). Since there's effort to create topic-specific instances and people's interests are often different, there arises a need in managing communities on other instances than your own. Of course, you can make your *main* account a moderator on a different instance (or so I think, I haven't tried yet), but this just means that the list of moderators gets cluttered and you have to manage a lot of accounts. I guess this is slightly related to the ["multireddit" topic](https://github.com/LemmyNet/lemmy-ui/issues/1113).

There are bound to be some things to consider on how to handle accounts linked to others from defederated instances. I'm just not able to think it this far through and most definitely will miss something important, but some way to manage multiple accounts would be nice for personal convenience and overall fediverse stability.

**Comment by AlexBlackRaven** [comment]
> What you're looking for might be "nomadic identity", where you just have one account that can work across multiple instances.
> 
> I believe it's somewhere down the line, but there's no concrete plan for implementation just yet.

Ah, thanks, I changed the title so it's more searchable. Good to know it's on the radar already, even if far down the line.

**Comment by zyansheep** [comment]
One way to do this would be to have universal identities (pub/priv keypair) and then record users' comment, post, and profile history to a distributed self-verifying user-append-only datastructure (essentially a per-user blockchain, append-able only with the user's private key) as seen in [scuttlebutt](https://ssbc.github.io/scuttlebutt-protocol-guide/) and [polycentric](https://docs.polycentric.io/#cryptography). Parts of this datastructure can be kept track of by different servers based on the posts / comments those servers care to store, or the user themselves to avoid losses of data.

Benefits of this system include:
 - Easily be able to see if a given server is censoring a user's post/comment/profile (because each block in the chain has a count, so if the server sends block 49 and block 51 but not block 50, you will know you are missing something).
 - Servers can't replicate someone's cryptographic signature (and lemmy clients could do signature verification to make sure)

Downsides:
 - If you lose your keys, you lose your account :(, maybe this could be solved via password-protected recovery data split among multiple trusted servers / other users?
 - If someone steals your keys, they can post as you or invalidate the existing public keys to a new set, this probably means that servers will have to step in and verify for the rest of the network which key is the legitimate one, this can probably be done through the existing account system.
 
Ultimately, this system would allow some of the burden of maintaining identity to be shifted from the servers to the users, while still allowing the servers to manage important parts like names, preventing impersonation and providing account recovery mechanisms.

This is a difficult problem, and I don't know how well it would work for lemmy (would likely need a sizable update to the server protocol and clients and managing the transition from server-based to cryptographic identities would be difficult), but it would definitely be a big step towards solving one of the major issues with federated protocols :D

**Comment by DHager** [comment]
It feels to me like "prove instance administrators aren't removing content" is an additional step beyond nomadic identity which brings in a _lot_ of complexity and resources.

In contrast, consider a much narrower use-case: "I posted a lot on an instance X, it shut down for funding reasons, now I'm on instance Y and I want to prove both accounts are myself."

One way to solve that would be if the user created each of the two identities in such a way that they are defined by the same public key, and then the private key can be used to sign content that they submit. Sure, if you lose your key you're stuck, but it'd still be a step up in control.

**Comment by Nutomic** [comment]
This would require backend changes, but its also too complex and out of scope for Lemmy.

**Comment by beedell-roke** [comment]
@Nutomic, I agree, especially regarding https://github.com/LemmyNet/lemmy-ui/issues/1571#issuecomment-1613641529. However, regarding the undermentioned:

> In contrast, consider a much narrower use-case: "I posted a lot on an instance X, it shut down for some reason, now I'm on instance Y and I want to prove I am the same owner of both accounts."

...this is what Mastodon provides: [^1]

<blockquote>

### Profile redirect

Redirecting your account disables posting from that account and displays a "profile moved" notice indicating your new account. Anyone viewing your profile can see this notice and will know to follow you at your new account. The redirect can be canceled at any time.

### Account aliases

Profile moves can only be initiated when your two accounts have been aliased. You will need to set your old account as an alias of your new account before initiating the move. Setting aliases is harmless and reversible on its own.

</blockquote>

Consequently, does anything track integrating this into Lemmy?

[^1]: [`github.com/mastodon/documentation/blob/04ecf2677ce1239e812e74a93c4b12a3f738cbd8/content/en/user/moving.md#profile-redirect-redirect`][1]

[1]: https://github.com/mastodon/documentation/blob/04ecf2677ce1239e812e74a93c4b12a3f738cbd8/content/en/user/moving.md#profile-redirect-redirect:~:text=redirection%20or%20migration.-,Profile%20redirect,-%7B%23redirect%7D&text=Redirecting%20your%20account%20disables%20posting%20from%20that%20account%20and%20displays%20a%20%22profile%20moved%22%20notice%20indicating%20your%20new%20account.%20Anyone%20viewing%20your%20profile%20can%20see%20this%20notice%20and%20will%20know%20to%20follow%20you%20at%20your%20new%20account.&text=exported%20CSV%20files.-,Account%20aliases,-%7B%23aliases%7D

**Comment by dessalines** [comment]
We have user exporting, but don't support any nomadic identity. If someone wants to work on it tho, they can make a back-end issue and I'll assign it to them.

**Comment by beedell-roke** [comment]
> We have user exporting

@dessalines, if you mean what I just mentioned, are you able to link to any documentation it? I've not been able to locate any, and have a lot of accounts across different instances that I'd like to link. I realise this is somewhat off-topic, so thanks for the indulgence.

**Comment by dessalines** [comment]
Go to your user settings page:

![Image](https://github.com/user-attachments/assets/7b23f56f-0acc-4e31-9440-ed662f4ac1f7)

**Comment by beedell-roke** [comment]
@dessalines, that's not equivalent. I'm referring to genuine *linkage*:

<img width="1018" height="1044" alt="Image" src="https://github.com/user-attachments/assets/c830ae7f-16d6-4a3f-9891-6713206c8154" />

**Comment by dessalines** [comment]
That seems like something you'd put in your bio.
